core/app/models/spree/product.rb

```diff
@@ -42,11 +42,24 @@ class Product < Spree::Base
     translates(*TRANSLATABLE_FIELDS)
 
     self::Translation.class_eval do
+      before_update :set_slug
       acts_as_paranoid
       # deleted translation values also need to be accessible for index views listing deleted resources
       default_scope { unscope(where: :deleted_at) }
+      def set_slug
+        if self.name.blank? & self.slug.blank?
```

Comment:
You should not use bit operators to do boolean logic like this. Stick to `&&`.

Reply:
ups, typo